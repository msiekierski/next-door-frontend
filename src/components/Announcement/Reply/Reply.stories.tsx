import React from "react";
import Reply from "./Reply";

export default {
  title: "Components/Post/Reply",
  components: Reply,
};

export const reply = () => <Reply />;
