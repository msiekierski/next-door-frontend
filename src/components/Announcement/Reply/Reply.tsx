import React, { FormEvent, FunctionComponent, useContext, useRef } from "react";
import { Button, FormControl, InputGroup } from "react-bootstrap";
import IComment from "../Comment/IComment";
import { IUser } from "../../Login/IUser";
import { UserContext } from "../../Login/UserContext";
import { createReply } from "../../../API/announcements";

interface OwnProps {
  commentAdd: Function;
  idAnnouncement: number;
}

type Props = OwnProps;

const Reply: FunctionComponent<Props> = ({ commentAdd, idAnnouncement }) => {
  const content = useRef<HTMLInputElement>(null);
  const user = useContext<IUser | null>(UserContext);

  async function handleReply(e: FormEvent) {
    e.preventDefault();
    console.log(idAnnouncement);
    let desc = content && content.current ? content.current.value : "";
    const newComment: IComment = {
      creationDate: new Date().toDateString() + " " + new Date().toLocaleTimeString(),
      description: desc,
      idAccount: user?.idAccount,
      idAnnouncement: idAnnouncement,
    };
    let res = await createReply(newComment);
    commentAdd(newComment);
  }

  return (
    <InputGroup>
      <FormControl placeholder="Put your comment here" aria-label="Put your comment here" ref={content} />
      <InputGroup.Append>
        <Button variant="primary" onClick={handleReply}>
          Reply
        </Button>
      </InputGroup.Append>
    </InputGroup>
  );
};

export default Reply;
