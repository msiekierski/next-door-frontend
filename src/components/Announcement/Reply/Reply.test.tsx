import React from "react";
import renderer from "react-test-renderer";
import Reply from "./Reply"

interface Props {
    commentAdd: Function;
    idAnnouncement: number;
}

const props:Props = {
    commentAdd: () => {},
    idAnnouncement: 1
}

it("renders correctly", () => {
  const tree = renderer.create(<Reply {...props} />);
  expect(tree).toMatchSnapshot();
});
