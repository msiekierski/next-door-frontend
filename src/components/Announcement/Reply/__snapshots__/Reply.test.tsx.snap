// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="input-group"
>
  <input
    aria-label="Put your comment here"
    className="form-control"
    placeholder="Put your comment here"
  />
  <div
    className="input-group-append"
  >
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reply
    </button>
  </div>
</div>
`;
